Mixmatch/streetview_dataset/parse_data_to_tfrecord_lib.py

```diff
@@ -0,0 +1,207 @@
+    examples = {0:[], 1:[]}
+    
+    for image_features in parsed_image_dataset:
+        img_name, confidence, bbox = parse_labels(image_features)
+        # The format fo the image_name is XXXXXX_Y.jpg, the Y represents the view of the image.
+        view = img_name.split('.')[0][-1]
+        keep_image = include_top_camera or (view != '5' and view != '0')
+        
+        if img_name and keep_image:
+            img_path = os.path.join(folder_path, img_name)
+            
+            if only_keep_top_confidence:
+                target = strip_top_confidence_bbox(confidence, bbox)
+            else:
+                target = strip_all_qualified_bbox(confidence, bbox)
+            
+            if not target:
+                continue
+
+            try:
+                img = Image.open(img_path, "r")
+            except Exception as e:
+                print(e)
+                print(img_path + " is not valid")
+                continue
+
+            # Exclude all non RGB images
+            if len(img.getbands()) != 3:
+                continue
+
+            for t in target:
+                crop_img = img
+                if 'bbox' in t:
+                    crop_img = crop_img.crop(t['bbox'])
+                crop_img = crop_img.resize(OUTPUT_IMAGE_SIZE)
+                example = img_to_example(crop_img, t['label'])
+                examples[t['label']].append(example)
+                
+    return examples
+
+# Write positive and negative tfexamples to tfrecord using the writer. A balance boolean parameter can decide to balance the pos and neg examples count.
+def write_tfexample_to_tfrecord(positive_examples, negative_examples, balance, writer):
+    take = float('inf')
+    num_pos = len(positive_examples)
+    num_neg = len(negative_examples)
+    
+    if balance:    
+        take = min(num_pos, num_neg)
+    
+    for i in range(min(num_pos, take)):
+        writer.write(positive_examples[i].SerializeToString())
+    
+    for i in range(min(num_neg, take)):
+        writer.write(negative_examples[i].SerializeToString())
+
+
+# Write tfrecords in batches of input record files.
+def batch_read_write_tfrecords(file_range, input_record_path, input_img_path, writer, detection_property):
+    include_top_camera = detection_property['include_top_camera']
+    only_keep_top_confidence = detection_property['only_keep_top_confidence']
+    balance = detection_property['balance']
+    pos_examples, neg_examples = [], []
+    
+    for i in range(file_range[0], file_range[1]):
+        file_name = "./streetlearn_detections_tfexample-" + str(i).zfill(5) + "-of-01000.tfrecord"
+        parsed_image_dataset = read_tfrecord(os.path.join(input_record_path, file_name))
+        examples = generate_tfexamples_from_detections(parsed_image_dataset, input_img_path, include_top_camera, only_keep_top_confidence)
+        neg_examples.extend(examples[0])
```

Comment:
Since you already made `examples` in a dict form, you can just pass `examples` to `write_tfexample_to_tfrecord` and address the dict format in the function description.

Reply:
Since I will generate examples in every for loop iteration, and I have to wait until every tfrecords is read to call the write_tfexample_to_tfrecord(). I think it is better for me to separate positive and negative results within every iteration. If not, there will be a list of examples dictionary generated. [examples_iter1, examples_iter2, ..., examples_iter_n]
Tell me if this makes sense to you.

Reply:
SGTM.